04-kogito-travel-agency/src/main/java/org/kie/kogito/app/VertxRouter.java

```diff
@@ -0,0 +1,37 @@
+package org.kie.kogito.app;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+import io.quarkus.vertx.web.Route;
+import io.vertx.core.http.HttpHeaders;
+import io.vertx.ext.web.RoutingContext;
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+
+import static io.vertx.core.http.HttpMethod.GET;
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+@ApplicationScoped
+public class VertxRouter {
+
+    @Inject
+    @ConfigProperty(name = "kogito.dataindex.url", defaultValue = "http://localhost:8180/graphql")
+    private String dataIndexURL;
+
+    @Route(path = "/", methods = GET)
+    public void handle(RoutingContext context) {
+        try {
+            String resource = context.vertx().fileSystem()
+                    .readFileBlocking("META-INF/resources/index.html")
+                    .toString(UTF_8)
+                    .replace("__GRAPHIQL_ENDPOINT__", "\"" + dataIndexURL + "\"");
```

Comment:
just wondering... should we do this only once?

Reply:
I mean the reading and replacing...

Reply:
true, Improved it a bit now.